src/test/run-pass/issue-17718-borrow-interior.rs

```diff
@@ -9,17 +9,13 @@
 // except according to those terms.
 
 struct S { a: usize }
+
 static A: S  = S { a: 3 };
 static B: &'static usize = &A.a;
-//~^ ERROR: cannot refer to the interior of another static
 static C: &'static usize = &(A.a);
-//~^ ERROR: cannot refer to the interior of another static
 
 static D: [usize; 1] = [1];
 static E: usize = D[0];
-//~^ ERROR: cannot refer to the interior of another static
-//~^^ ERROR: cannot refer to other statics by value
 static F: &'static usize = &D[0];
-//~^ ERROR: cannot refer to the interior of another static
 
 fn main() {}
```

Comment:
I think it would be good to add some runtime tests here that assert all the statics' values in order to make sure that the right fields have been read